packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx

```diff
@@ -11,13 +12,17 @@ import { ThreadPreview } from './thread_preview';
 import { Footer } from '../../footer';
 
 type DiscussionsPageProps = {
-  topicName?: string;
-  stageName?: string;
+  topic?: string;
+  stage?: string;
 };
 
-const DiscussionsPage = ({ topicName, stageName }: DiscussionsPageProps) => {
+const DiscussionsPage = ({ topic, stage }: DiscussionsPageProps) => {
   const [threads, setThreads] = useState([]);
   const [initializing, setInitializing] = useState(true);
+  const params = useParams();
+  const topicName = params.topic;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [stageName, setStageName] = useState<string>(searchParams.get('stage'));
```

Comment:
`setSearchParams` and `setStageName` are not used. But anyways, why do we need to create state for them if there are not mutable here?

Reply:
Both removed.